Assets/MirageSDK/Demo/Scripts/DemoScript.cs

```diff
@@ -0,0 +1,197 @@
+				await _contractHandler.MintItems();
+				await LoadCharacter();
+			}
+
+			await CheckCharactersEquippedHatAndDisplay();
+			await GetItemTokensBalanceAndUpdateInventory();
+
+			var characterID = await _contractHandler.GetCharacterTokenId();
+			var equippedHatID = await _contractHandler.GetHat();
+			UpdateUILogs("characterID: " + characterID + " / HatID: " + equippedHatID);
+
+			var isApprovedForAll = await _contractHandler.CheckIfCharacterIsApprovedForAll();
+			if (!isApprovedForAll)
+			{
+				await _contractHandler.ApproveAllForCharacter(true);
+			}
+
+			/*_source = new CancellationTokenSource();
+			CancellationToken token = _source.Token;
+			CheckForChangesInEquippement(token).Forget();*/
+		}
+		
+		private async UniTaskVoid CheckForChangesInEquippement(CancellationToken token)
+		{
+			while (!token.IsCancellationRequested)
+			{
+				await CheckCharactersEquippedHatAndDisplay();
+				await GetItemTokensBalanceAndUpdateInventory();
+				await UniTask.Delay(RefreshTimer, cancellationToken: token);
+			}
+			
+		}
+
+		private GameObject CreateHat(ItemDescription item)
+		{
+			var prefab = item.GameObjectPrefab;
+			var instantiatedGO = Instantiate(prefab, _characterHead.transform, false);
+			instantiatedGO.SetActive(false);
+			return instantiatedGO;
+		}
+
+		private void OnButtonClick(string address)
+		{
+			OnButtonClickAsync(address).Forget();
+		}
+
+		private async UniTask OnButtonClickAsync(string address)
+		{
+			await EquipHat(address);
+		}
+
+		private async UniTask CheckCharactersEquippedHatAndDisplay()
+		{
+			var equippedHat = await _contractHandler.GetHat();
+			if (equippedHat.TryConvertToHatColour(out var hatColour))
+			{
+				UpdateHatVisuals(hatColour);
+			}
+			else
+			{
+				RemoveHatVisuals();
+			}
+		}
+
+		private async UniTask EquipHat(string address)
+		{
+			await _contractHandler.ChangeHat(address);
+			await CheckCharactersEquippedHatAndDisplay();
+			await GetItemTokensBalanceAndUpdateInventory();
+		}
+
+		private void UpdateHatVisuals(HatColour hatColour)
+		{
+			foreach (var item in _items)
+			{
+				var isRightColour = hatColour == item.Key;
+				item.Value.GameObject.SetActive(isRightColour);
+			}
+		}
+
+		private void RemoveHatVisuals()
+		{
+			foreach (var item in _items.Values)
+			{
+				item.GameObject.SetActive(false);
+			}
+		}
+
+		private async UniTask<bool> LoadCharacter()
+		{
+			var tokenID = await _contractHandler.GetCharacterTokenId();
+
+			if (tokenID == -1)
+			{
+				return false;
+			}
+
+			_characterID.text = tokenID.ToString();
+			_character.SetActive(true);
+
+			return true;
+		}
+
+		private async UniTask GetItemTokensBalanceAndUpdateInventory()
+		{
+			for (var i = 0; i < _itemsDescriptions.Descriptions.Count; i++)
+			{
+				var addressTokenBalance =
+					await _contractHandler.GetItemBalance(_itemsDescriptions.Descriptions[i].Address);
+				_inventory.ShowInventoryItem(i, addressTokenBalance >0 , addressTokenBalance);
```

Comment:
format your code pleas.
```suggestion
				_inventory.ShowInventoryItem(i, addressTokenBalance > 0 , addressTokenBalance);
```